russstation/code/game/objects/items/miscellaneous.dm

```diff
@@ -0,0 +1,172 @@
+//constructing a wetmore slippery sign:
+//obtain a wet floor sign (from cargo, chemistry(plastic), or your vendor )
+//apply floor buffer to sign (from RND)
+//apply prox sensor to sign (from robo/cargo)
+
+/obj/item/clothing/suit/caution/attackby(obj/item/I, mob/living/user)
+	. = ..()
+	var/atom/L = drop_location()
+	var/datum/quirk/family_heirloom/heirloomCheck 
+
+	//checks all their quirks for the family heirloom quirk- so we can check if they have a wet floor sign heirloom later 
+	for(var/datum/quirk/hasQuirk in user.roundstart_quirks)
+		if(istype(hasQuirk, /datum/quirk/family_heirloom))
+			heirloomCheck = hasQuirk
+			break
+
+	if(istype(I, /obj/item/janiupgrade))
+		if(src == heirloomCheck.heirloom) //make sure jannies don't accidentally use their family heirlooms
+			to_chat(user, "<span class='warning'>You wouldn't want to tamper with your heirloom [src.name]!<span>")
+		else
+			to_chat(user, "<span class='notice'>You add a [I.name] to the bottom of the [src.name].<span>")
+			qdel(I)
+			qdel(src)
+			new /obj/item/clothing/suit/caution/incomplete(L, 1)
+
+		
+/obj/item/clothing/suit/caution/incomplete
+	name = "Incomplete wet floor sign"
+	slot_flags = null
+
+/obj/item/clothing/suit/caution/incomplete/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>The [src.name] has a floor buffer underneath. You could <i>attach a sensor</i> to it, or <i>undo the screws</i> to remove it.</span>"
+
+/obj/item/clothing/suit/caution/incomplete/attackby(obj/item/I, mob/living/user)
+	. = ..()
+	var/atom/L = drop_location()
+
+	if(istype(I, /obj/item/screwdriver))
+		to_chat(user, "<span class='notice'>You detatch the floor buffer from the [src.name].<span>")
+		qdel(src)
+		new /obj/item/clothing/suit/caution(L, 1)
+		new /obj/item/janiupgrade(L, 1)
+
+	if(istype(I, /obj/item/assembly/prox_sensor))
+		to_chat(user, "<span class='notice'>You add a [I.name] to the floor buffer on the [src.name].<span>")
+		qdel(I)
+		qdel(src)
+		var/obj/item/S = /obj/item/clothing/suit/caution/slippery
+		if(prob(1)) //1% chance for a name
+			S.name = pick("Ms. Lippy", "Mr. Walky", "Monitor Hallsky")
```

Comment:
Would this be changing only items created under the alias of `/obj/item/S` or would this change all `/obj/item/clothing/suit/caution/slippery`s?